kube-runtime/src/watcher.rs

```diff
@@ -640,19 +640,58 @@ pub fn watch_object<K: Resource + Clone + DeserializeOwned + Debug + Send + 'sta
 
 /// Default watch [`Backoff`] inspired by Kubernetes' client-go.
 ///
-/// Note that the exact parameters used herein should not be considered stable.
-/// The parameters currently optimize for being kind to struggling apiservers.
-/// See [client-go's reflector source](https://github.com/kubernetes/client-go/blob/980663e185ab6fc79163b1c2565034f6d58368db/tools/cache/reflector.go#L177-L181)
-/// for more details.
+/// This fn has been moved into [`DefaultBackoff`].
 #[must_use]
+#[deprecated(
+    since = "0.84.0",
+    note = "replaced by `watcher::DefaultBackoff`. This fn will be removed in 0.88.0."
+)]
 pub fn default_backoff() -> impl Backoff + Send + Sync {
-    let expo = backoff::ExponentialBackoff {
+    let bo = default_exponential_backoff();
+    ResetTimerBackoff::new(bo, DEFAULT_RESET_DURATION)
+}
+
+/// Default watcher backoff inspired by Kubernetes' client-go.
+///
+/// The parameters currently optimize for being kind to struggling apiservers.
+/// The exact parameters are taken from
+/// [client-go's reflector source](https://github.com/kubernetes/client-go/blob/980663e185ab6fc79163b1c2565034f6d58368db/tools/cache/reflector.go#L177-L181)
+/// and should not be considered stable.
+///
+/// This struct implements [`Backoff`] and is the default strategy used
+/// when calling `WatchStreamExt::default_backoff`. If you need to create
+/// this manually then [`DefaultBackoff::default`] can be used.
+pub struct DefaultBackoff(Strategy);
+
+struct Strategy(ResetTimerBackoff<ExponentialBackoff>); // encapsulate strategy
+
+fn default_exponential_backoff() -> backoff::ExponentialBackoff {
```

Comment:
Can be moved into the `Default` impl of `DefaultBackoff` after changing the `default_backoff` signature